### ISSUE TITLE
species ID for combined model hardwired in ML data import

### ISSUE THREAD

**Issue by mkclapp** [calacademy-research/COMB#53]
When we first developed this script, we used a subset of the ML data that was manually filtered to RBNU. The current script reads in that filtered dataframe, but I propose the following:

renaming the script to singlespp_combined to distinguish it from (future) community models
<s>reading in the full ML output (the one that contains logits for all species) and using the speciesCode object that's set in line 19 to filter to the species of interest</s>  **Update** Matt and I have re-engineered this part of the code so this will be tackled in a separate issue 
<s>saving the output of that model with a name unique to that species so that model outputs for multiple species can be saved and compared</s> this will be wrapped into the above "API"

**Comment by matth79** [comment]
Yes, this is "the" critical path issue. I'll leave it to you, since you are reporter and assignee.

- Rename sounds good.
- Reading the full ML output is probably a good place to start, but I can see a case for investing some time into making a library that could efficiently read a data.table for only a specified subset of species and only a specified subset of years.
  - "library" seems appealing to me because I anticipate different variants of model fitting scripts will all want more or less than same "readML" but maybe with some different parameters.
  - One but not the only possibility is to split dataML.csv into a filesystem hierarchy {species}/{point}/{year}/logits.csv, where logits.csv has just (filename, start_time, logit).
  - I could contribute some help on this part if you'd like.
- Saving output sounds good, although I will have to wait to learn what and how, since combined.R doesn't do it yet, and I haven't studied the other scripts.

**Comment by mkclapp** [comment]
Hi Matt, thanks for the responses.

> Yes, this is "the" critical path issue. I'll leave it to you, since you are reporter and assignee.

Aye aye! 

> 
> * Rename sounds good.
> * Reading the full ML output is probably a good place to start, but I can see a case for investing some time into making a library that could efficiently read a data.table for only a specified subset of species and only a specified subset of years.
>   
>   * "library" seems appealing to me because I anticipate different variants of model fitting scripts will all want more or less than same "readML" but maybe with some different parameters.
>   * One but not the only possibility is to split dataML.csv into a filesystem hierarchy {species}/{point}/{year}/logits.csv, where logits.csv has just (filename, start_time, logit).
>   * I could contribute some help on this part if you'd like.

Agreed -- we will be spending quite a bit of time on different combinations of year, species, and logit cutoffs while we build and optimize models, so having a library script that could flexibly handle this would be good. I am running into some issues on readML.subset.R (which is where I believe the ML output subsetted to rebnut originated) but to fix them would entail making some decisions about how we ingest/digest raw ML output. I have some time tomorrow afternoon for a Meet if you do...

> * Saving output sounds good, although I will have to wait to learn what and how, since combined.R doesn't do it yet, and I haven't studied the other scripts.

This was my original intent today, to save the output to models/output and then write a new script within `models` that inspects and visualizes it (or any jags output object). It's trivial (just writing the output to file) but it of course brings up the larger issue of if/whether we want to generate these outputs from scratch every time we need them (costly time- and RAM-wise) or to save them somewhere and deal with storing them/git ignore/version control.

**Comment by ddkapan** [comment]
Great conversation, a simple first step is to make a function to do the filtering, filter_ml <- function(x, y, z ...){ function guts}

Then a simple call is 

>dataML %>% 
  filter_ml(x, y, z ...) %>%
  model_stuff_here ...

Once it works for one, it works for all, see purr::map() class of functions and even better furrr drop in parallel replacements ... then you can iterate across species and study designs etc!!!

:D

**Comment by mkclapp** [comment]
status update: since we've opened a can of worms re: data ingestion, I am stashing my changes until tomorrow when Matt and I can meet to discuss a better structure for the data library.

**Comment by matth79** [comment]
I sent a calendar invite to Mary at an arbitrary time tomorrow afternoon. Feel free to move it if it doesn't work. My schedule is flexible.

My opinion is it's always fine to do something "quick and dirty" with the plan to fix later. (as long as it really is quick)
